ansible_collections/arista/avd/roles/eos_designs/templates/network_services/logic.j2

```diff
@@ -0,0 +1,34 @@
+
+{% set tmp_tenants = [] %}
+{# tenants #}
+{% for tenant in tenants | arista.avd.convert_dicts | arista.avd.natural_sort('name') %}
+{%     if tenant.name in switch.tenants | arista.avd.default([]) %}
+{%         set switch_tenant = switch.tenants[tenant.name] %}
+{#         vrfs #}
+{%         set tmp_vrfs = [] %}
+{%         for vrf in tenant.vrfs | arista.avd.convert_dicts | arista.avd.natural_sort('name') %}
```

Comment:
1) I don't know the reason, why we sort the data in eos_designs, because we sort them anyway in eos_cli_config_gen.
2) If you think sorting is required then don't sort them here. Just append them to the tmp_list now and sort that tmp_list when you are using it. Because tmp_list mostly will have few items compared to the actual parent list (this one). Also sometimes the if check might fail for all entries leading to an empty tmp_list. So in that case we are wasting resources to sort the list unnecessarily.

Reply:
We will reference these variables in many places, so sorting them every time is ineffective. We _could_ add a final sort before writing the variable, instead of sorting on the input, but I did not see any measurable difference. Also the protection builtin to the `natural_sort` filter saves us another `default([])`

We need to sort the data in eos_designs primarily to avoid any changes. The documentation and the csv files changed when I didn't sort. I wanted to avoid _any_ changes.

Reply:
In general the "structured config" is also a "product" which is used by many other areas, so having it sorted and nice makes it easier for the next role.

Reply:
Got it, thanks.